r9k.py

```diff
@@ -21,19 +22,34 @@ async def handle_message(self, bot, message):
         if len(message.content) == 0:
             return
 
+        # Check for text in message
         h = hashlib.sha256(message.content.lower().encode('utf-8')).hexdigest()
+        h_exists = h in self.cache
+        # Check for image attached to message
+        img_h = None
+        img_h_exists = False
+        if message.attachments != []:
+            file = message.attachments[0]
+            if file.filename[file.filename.rfind("."):] in [".jpg",".JPG",".png",".PNG",".gif",".GIF"]:
+                # Discord API Attachment.read() returns bytes
+                file_bytes = await file.read()
+                img_h = hashlib.sha256(file_bytes.hexdigest())
+                img_h_exists = img_h in self.cache

```

Comment:
Perhaps we switch this to a hash check to a hash similarity check and add some threshold. since even two images of the same thing will probably vary a little due to lossy compression or slightly different crops etc? Not sure how viable it is though.